### PR TITLE
Adds more unit tests for improper cables

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Adds more unit tests for improper cables, such as duplicate cables on top of each other, or cables that aren't connected to another cable in the direction they're pointing.

Also removes GetOppositeDir because it was a duplicate proc of reverse_direction

Prequestites:
Cyberiad: #24266
Delta: #24267
Meta: #24265
Farragus: #24272

## Why It's Good For The Game
Makes mapping better

## Testing

CI will not pass first try.

## Changelog
NPFC

### CODE REVIEW

File: code/modules/unit_tests/map_tests.dm [hunk 1]
```diff
@@ -124,3 +124,35 @@
...
+			var/turf/secondary = get_step(T, cable.d1)
+			for(var/obj/structure/cable/other_cable in secondary.contents)
+				if(reverse_direction(cable.d1) == other_cable.d1 || reverse_direction(cable.d1) == other_cable.d2)
+					return
+			Fail(T, "tile has an unconnected cable (d2 connection: [uppertext(dir2text(cable.d2))]).")
+
+/datum/map_per_tile_test/duplicate_cable_check
+
+/datum/map_per_tile_test/duplicate_cable_check/CheckTile(turf/T)
+	var/obj/structure/cable/cable locate() in T.contents
```
Comment (GDNgit) [inline code/modules/unit_tests/map_tests.dm:153]:
```suggestion
	var/obj/structure/cable/cable = locate() in T.contents
```

**Review by S34NW — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]
